sematic/config/settings.py

```diff
@@ -60,36 +239,63 @@ def _normalize_enum(enum_type: EnumType, obj: Any):  # type: ignore
         return obj
 
     try:
-        return enum_type[str(obj)]  # type: ignore
+        return enum_type[str(obj)]
     except KeyError:
         return None
 
 
-def _load_settings(file_path: str) -> Dict[str, Union[str, Dict[str, str]]]:
+def _load_settings(file_path: str, vars: Type[AbstractSettingsVar]) -> Settings:
     """
     Loads the settings from the specified settings file, returning their raw string
     representations, or an empty dict if the file is not found.
     """
+    settings = Settings()
+
     try:
         with open(file_path, "r") as f:
-            return yaml.load(f, yaml.Loader)
+            settings = yaml.load(f, yaml.Loader)
 
     except FileNotFoundError:
         logger.debug("Settings file %s not found", file_path)
-        return {}
 
+    if settings is None:
+        settings = Settings()
+
+    if DEFAULT_PROFILE not in settings:
+        settings[DEFAULT_PROFILE] = ProfileSettings()
+
+    # Normalizing settings to only have expected values and format as enum
+    for profile_settings in settings.values():
+        # Doing this instead of usual .items iteration because we mutate the keys
+        # as we go
+        for key in list(profile_settings):
+            normalized_key = _normalize_enum(vars, key)
+
+            if normalized_key is not None:
+                profile_settings[normalized_key] = profile_settings[key]
```

Comment:
If the user manually adds a setting to the file with a typo in its name, this code will silently ignore it, leading to the user to conclude that the feature they are trying to use does not work.

Please add a warning log that an unknown setting was encountered.

Reply:
Done.